pkg/daemon/sync.go

```diff
@@ -125,13 +113,86 @@ func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string
 	return err
 }
 
+// getLastResources loads last-synced resources
+func (d *Daemon) getLastResources(ctx context.Context, rat ratchet) (map[string]resource.Resource, error) {
+	lastResources := rat.CurrentResources()
+	if lastResources != nil {
+		return lastResources, nil
+	}
+
+	currentRevision, err := rat.CurrentRevision(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Repo has never been cloned yet
+	if currentRevision == "" {
+		return make(map[string]resource.Resource), nil
+	}
+
+	// First sync
+	d.Logger.Log("info", "resource map is not initialized, despite repo has been cloned")
```

Comment:
Since we're no longer thinking of this as an unexpected situation, it doesn't need a log message. But if you still want one, I would have a message like "initialising last sync resources from last revision". The "despite" makes it sound exceptional, but it will happen every time fluxd starts.

Reply:
I agree. Removed the log. Thank you!